### ISSUE TITLE
querying uncovered index column on a non-existent row causes crash in yb::ql::RowsResult::SetPagingState()

### ISSUE THREAD

**Issue by kmuthukk** [YugaByte/yugabyte-db#670]
Test Setup:

```
CREATE TABLE IF NOT EXISTS kv (k int PRIMARY KEY, v int, c text) WITH TRANSACTIONS = { 'enabled' : true };

CREATE INDEX by_v ON kv(v);

INSERT INTO kv(k, v, c) VALUES (1, 11, 'a');
INSERT INTO kv(k, v, c) VALUES (2, 22, 'b');
```

Querying for columns covered by the index (such as k which is automatically in the index) on a non-existent row works fine and returns 0 rows as expected.
```
cqlsh:demo> SELECT k FROM kv WHERE v=33;

 k
---

(0 rows)
```

But querying for a column which is not covered by the index for a non-existent row results in the following crash, instead of an empty result set.
```
cqlsh:demo> SELECT c FROM kv WHERE v=33;
OperationTimedOut: errors={'127.0.0.1': 'Client request timeout. See Session.execute[_async](timeout)'}, last_host=127.0.0.1
```

**Comment by robertpang** [comment]
Resolved in commit [46d73fc](https://github.com/YugaByte/yugabyte-db/commit/46d73fc3218865b14595ec5e5b9fa269633df21e).

**Comment by hreidar** [comment]
Hi, @robertpang, is it possible that this bug is resurfacing in version 1.2.0.1-b3 ?

```
cqlsh> create keyspace kea150
   ...   with replication = {
   ...     'class' : 'SimpleStrategy',
   ...     'replication_factor' : 1
   ...   };
cqlsh>
cqlsh> use kea150;
cqlsh:kea150>
cqlsh:kea150> CREATE TABLE IF NOT EXISTS lease6_types (
          ...     lease_type int,                             -- Lease type code.
          ...     name varchar,                               -- Name of the lease type
          ...     PRIMARY KEY ((lease_type))
          ... ) with transactions = { 'enabled' : true };
cqlsh:kea150> 
cqlsh:kea150> INSERT INTO lease6_types (lease_type, name) VALUES (0, 'IA_NA');
OperationTimedOut: errors={'127.0.0.1': 'Client request timeout. See Session.execute[_async](timeout)'}, last_host=127.0.0.1
cqlsh:kea150>
```
